test/Fencer/Rules/Test.hs

```diff
@@ -134,7 +151,22 @@ test_rulesLoadRulesRecursively =
         [ ("domain1" </> "config.yml", domain1Text)
         , ("domain2" </> "config" </> "config.yml", domain2Text) ]
       )
-      (#result [domain1, domain2])
+      (#result $ Success [domain1, domain2])
+
+-- | Test that 'loadRulesFromDirectory' returns exceptions for an
+-- invalid domain.
+test_rulesLoadRulesException :: TestTree
+test_rulesLoadRulesException =
+  testCase "Rules fail to load for an invalid domain" $
```

Comment:
This should clarify that rules fail to load even if just one domain is invalid.

Reply:
I extended the documentation for this test function. Does that address your concern?